controllers/client/grafana_client.go

```diff
@@ -142,6 +142,7 @@ func NewGeneratedGrafanaClient(ctx context.Context, c client.Client, grafana *v1
 	}
 
 	transport := NewInstrumentedRoundTripper(grafana.IsExternal(), tlsConfig, metrics.GrafanaApiRequests.MustCurryWith(prometheus.Labels{
+		"namespace":     grafana.Namespace,
```

Comment:
I'm not sure if we can use `namespace` label. - Normally, when Prometheus fetches metrics, it enriches metric metadata with k8s-specific context including `namespace` label, so it's likely to get overridden.
I've seen that some projects use `exported_namespace` for such scenarios, it's worth checking if there's a consensus / guidelines in Prometheus community on how to name this label.

Reply:
Good catch, It's very likely that the `namespace` label is immediately overwritten.
Since we already use `instance_name` would `instance_namespace` make sense to keep it consistent?

Reply:
I think the suggested naming is fine, yes.